systemd_ctypes/bus.py

```diff
@@ -320,10 +326,18 @@ def add_match(self, rule, handler):
     def add_object(self, path, obj):
         slot = Slot(obj.message_received)
         super().add_object(byref(slot), path, slot.callback, slot.userdata)
+        obj.registered_on_bus(self, path)
         return slot
 
 
 class BaseObject:
+    _bus = None
+    _path = None
+
+    def registered_on_bus(self, bus, path):
+        self._bus = bus
```

Comment:
These should get a `_dbus_` prefix, I think.

Reply:
Also: the way this is structured seems to assume that (a) objects only get published on one bus, and at a single path; and (b) are never unregistered.  Maybe we should have some checks around that.

Reply:
Registering an object a second time could automatically unregister it from its first place.